controllers/hetznercluster_controller_test.go

```diff
@@ -848,90 +848,102 @@ func TestSetControlPlaneEndpoint(t *testing.T) {
 		if (hetznerCluster.Spec.ControlPlaneLoadBalancer.Enabled == false){
-			t.Fatal("Wrong input for hetznerCluster.Spec.ControlPlaneLoadBalancer.Enabled. Should be true.")
+			t.Fatalf("Wrong value for ControlPlaneLoadBalancer.Enabled. Got: %v, Want: true", hetznerCluster.Spec.ControlPlaneLoadBalancer.Enabled)
 		}
 		if (hetznerCluster.Status.ControlPlaneLoadBalancer.IPv4 != "xyz"){
-			t.Fatal("Wrong input for hetznerCluster.Status.ControlPlaneLoadBalancer.IPv4")
+			t.Fatalf("Wrong input for hetznerCluster.Status.ControlPlaneLoadBalancer.IPv4. Got: %s, Want: xyz", hetznerCluster.Status.ControlPlaneLoadBalancer.IPv4)
 		}
 		// Values of hetznerCluster.Spec.ControlPlaneEndpoint.Host and hetznerCluster.Spec.ControlPlaneEndpoint.Port should change after execution of the function SetControlPlaneEndpoint()
-		// They should be the same as hetznerCluster.Status.ControlPlaneLoadBalancer.IPv4 for Host (Spec.ControlPlaneEndpoint.Host) and hetznerCluster.Spec.ControlPlaneLoadBalancer.Port for the Port (Spec.ControlPlaneEndpoint.Port)
+		// They should be the same as hetznerCluster.Status.ControlPlaneLoadBalancer.IPv4 for Host (Spec.ControlPlaneEndpoint.Host) and hetznerCluster.Spec.ControlPlaneLoadBalancer.Port for Port (Spec.ControlPlaneEndpoint.Port)
 		if (hetznerCluster.Spec.ControlPlaneEndpoint.Host != hetznerCluster.Status.ControlPlaneLoadBalancer.IPv4){
-			t.Fatal("Wrong value for Host set")
+			t.Fatalf("Wrong value for Host set. Got: %s, Want: %s", hetznerCluster.Spec.ControlPlaneEndpoint.Host, hetznerCluster.Status.ControlPlaneLoadBalancer.IPv4)
 		}
 		if (hetznerCluster.Spec.ControlPlaneEndpoint.Port != int32(hetznerCluster.Spec.ControlPlaneLoadBalancer.Port)){
-			t.Fatal("Wrong values for Port are set")
+			t.Fatalf("Wrong value for Port set. Got: %d, Want: %d", hetznerCluster.Spec.ControlPlaneEndpoint.Port, int32(hetznerCluster.Spec.ControlPlaneLoadBalancer.Port))
 		}
 		assert.Equal(t, true, got)
 	})
 	t.Run("function should return true if hetznerCluster.Spec.ControlPlaneLoadBalancer.Enabled is false but hetznerCluster.Spec.ControlPlaneEndpoint is not nil", func(t *testing.T) {
 		hetznerCluster := &infrav1.HetznerCluster{Spec: infrav1.HetznerClusterSpec{ControlPlaneLoadBalancer: infrav1.LoadBalancerSpec{Enabled: false}, ControlPlaneEndpoint: &clusterv1.APIEndpoint{Host: "xyz", Port: 1}}}
 		got := SetControlPlaneEndpoint(hetznerCluster)
 		if (hetznerCluster.Spec.ControlPlaneLoadBalancer.Enabled == true){
-			t.Fatal("Wrong input for hetznerCluster.Spec.ControlPlaneLoadBalancer.Enabled. Should be false.")
+			t.Fatalf("Wrong value for ControlPlaneLoadBalancer.Enabled. Got: %v, Want: false", hetznerCluster.Spec.ControlPlaneLoadBalancer.Enabled)
 		}
 		if (hetznerCluster.Spec.ControlPlaneEndpoint == nil){
-			t.Fatal("hetznerCluster.Spec.ControlPlaneEndpoint should not be nil")
+			t.Fatalf("Wrong input for hetznerCluster.Spec.ControlPlaneEndpoint. Got: %s, Want: (Host: 'xyz', Port: 1)", hetznerCluster.Spec.ControlPlaneEndpoint)
```

Comment:
you can leave it at the original "hetznerCluster.Spec.ControlPlaneEndpoint should not be nil"